mlflow/pyfunc/model.py

```diff
@@ -346,6 +354,75 @@ def predict_stream(
         )
 
 
+@experimental
+class ChatAgent(PythonModel, metaclass=ABCMeta):
+    """
+    A subclass of :class:`~PythonModel` that makes it more convenient to implement agents
+    that are compatible with popular LLM chat APIs. By subclassing :class:`~ChatAgent`,
+    users can create MLflow models with a ``predict()`` method that is more convenient
+    for chat tasks than the generic :class:`~PythonModel` API. ChatAgents automatically
+    define input/output signatures and an input example, so manually specifying these values
+    when calling :func:`mlflow.pyfunc.save_model() <mlflow.pyfunc.save_model>` is not necessary.
+
+    See the documentation of the ``predict()`` and ``predict_stream()`` method below for details on
+    that parameters and outputs that are expected by the ``ChatAgent`` API.
+
+    Before logging, ``predict()`` and ``predict_stream()`` methods only take in Pydantic models.
+    After logging, you can either pass one dictionary that conforms to a ChatAgentRequest schema or
+    two dictionaries for `messages` and `params`. Look at CHAT_AGENT_INPUT_EXAMPLE in mlflow.types.
```

Comment:
I thought about it again, passing dictionary to params field should not work, since the params schema is None and we will drop whatever parameters passed because of this